pkg/builder/builder.go

```diff
@@ -33,49 +33,91 @@ func NewBuilder(db db.DB, meta db.Client) Builder {
 	}
 }
 
-func (b *Builder) Build(cacheDir string) error {
-	indexDir := filepath.Join(cacheDir, "indexes")
+func (b *Builder) Build(indexDir string) error {
+	slog.Info("Building the index database")
 	count, err := fileutil.Count(indexDir)
 	if err != nil {
 		return xerrors.Errorf("count error: %w", err)
 	}
 	bar := pb.StartNew(count)
-	defer slog.Info("Build completed")
+	defer slog.Info("Build completed", slog.Int("count", count))
 	defer bar.Finish()
 
-	var indexes []types.Index
-	if err := fileutil.Walk(indexDir, func(r io.Reader, path string) error {
-		index := &crawler.Index{}
-		if err := json.NewDecoder(r).Decode(index); err != nil {
-			return xerrors.Errorf("failed to decode index: %w", err)
-		}
-		for _, ver := range index.Versions {
-			indexes = append(indexes, types.Index{
-				GroupID:     index.GroupID,
-				ArtifactID:  index.ArtifactID,
-				Version:     ver.Version,
-				SHA1:        ver.SHA1,
-				ArchiveType: index.ArchiveType,
-			})
+	err = fileutil.Walk(indexDir, func(r io.Reader, path string) error {
+		// 	// Process only TSV files
+		if filepath.Ext(path) != ".tsv" {
+			return nil
 		}
-		bar.Increment()
+		defer bar.Increment()
+
+		// Create a CSV reader for TSV format
+		reader := csv.NewReader(r)
+		reader.Comma = '\t' // Use tab as delimiter
+		reader.FieldsPerRecord = 5
+		reader.ReuseRecord = true // Reuse memory for performance
```

Comment:
We create same reader in crawler.
Perhaps we want to move this to create CSV reader in one place (and change settings (if needed) in one place)

Reply:
like https://github.com/aquasecurity/trivy-java-db/pull/57/commits/faee88925a001c49dac83992f0412326bcde9da4?

Reply:
yeah, looks good 👍